main/Main.hs

```diff
@@ -87,19 +89,31 @@ data Target = Target
   }
 
 -- | Recursively collect nix files in a directory
-collectNixFiles :: FilePath -> IO [FilePath]
+collectNixFiles :: FilePath -> StateT Bool IO [FilePath]
 collectNixFiles path = do
-  dir <- doesDirectoryExist path
+  dir <- lift $ doesDirectoryExist path
   if
     | dir -> do
-        files <- listDirectory path
+        warnDeprecated
+        files <- lift $ listDirectory path
         concat <$> mapM collectNixFiles ((path </>) <$> files)
     | ".nix" `isSuffixOf` path -> pure [path]
-    | otherwise -> pure []
+    | otherwise -> do
+        warnDeprecated
+        pure []
+  where
+    warnDeprecated = do
+      warningPrinted <- get
+      -- We don't want to print the warning more than once
+      unless warningPrinted $ do
+        lift $ hPutStrLn stderr $ "\ESC[33mPassing directories or non-Nix files (such as \"" <> path <> "\") is deprecated and will be unsupported soon, please use https://treefmt.com/ instead, e.g. via https://github.com/numtide/treefmt-nix\ESC[0m"
```

Comment:
I propose to point this to https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ni/nixfmt-tree/package.nix . That seems to just work as a drop in replacement and has treefmt already configured. If you just run treefmt plain it wants you to create a config.

```
treefmt pkgs/by-name/pa/paperless-ngx/
Error: failed to find treefmt config file: could not find [treefmt.toml .treefmt.toml] in /home/sandro/src/nixpkgs
```

Reply:
@infinisil kindly already did this in https://github.com/NixOS/nixfmt/commit/29ef29e01b3a6d72cc2f41890e426d9400416a35 =)